nixbase32/nixbase32.go

```diff
@@ -0,0 +1,134 @@
+/*
+Package nixbase32 implements the slightly odd "base32" encoding that's used
+in Nix.
+
+Nix uses a custom alphabet. Contrary to other implementations (RFC4648),
+encoding to "nix base32" also reads in characters in reverse order (and
+doesn't use any padding), which makes adopting encoding/base32 hard.
+*/
+package nixbase32
+
+import (
+	"fmt"
+	"strings"
+)
+
+// alphabet contains the list of valid characters for the Nix base32 alphabet.
+const alphabet = "0123456789abcdfghijklmnpqrsvwxyz"
```

Comment:
This was exported because it's used in the regex in the nixpath package.

Reply:
Understood, but that appears to be the only reason? As per the [Go proverb](https://go-proverbs.github.io/), "a little copying is better than a little dependency." I've responded more on the `nixpath` package comment to evaluate whether that regex is necessary.